server/src/test/java/org/opensearch/cluster/metadata/MetadataCreateIndexServiceTests.java

```diff
@@ -1337,7 +1332,7 @@ public void testRemoteStoreDisabledByUserIndexSettings() {
             assertThat(validationErrors.size(), is(1));
             assertThat(
                 validationErrors.get(0),
-                is(String.format(Locale.ROOT, "expected [%s] to be private but it was not", SETTING_REMOTE_STORE_ENABLED))
+                is(String.format(Locale.ROOT, "private index setting [%s] can not be set explicitly", SETTING_REMOTE_STORE_ENABLED))
```

Comment:
why is the error message changing?

Reply:
This is because we moved the settings from `FEATURE_FLAGGED_INDEX_SETTINGS` to `BUILT_IN_INDEX_SETTINGS` now. This was done in accordance with the javadoc for `FEATURE_FLAGGED_INDEX_SETTINGS`:

https://github.com/opensearch-project/OpenSearch/blob/951f051580a9feef37affd409d5892f50e9585a4/server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java#L233-L238